### PR TITLE
[W6.2b][T16-1]Nur Tristy Erna Binte Mohammed Harlan

### PR DESCRIPTION
**Pull request by tristyxxnana**
Add a method boolean isMutating() to the Command class. This method will return true for command types that mutate the data. e.g. AddCommand, DeleteCommand

### CODE REVIEW

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -66,4 +66,9 @@ public CommandResult execute() {
         }
     }
 
+    @Override
+    public boolean isMutating() {
```
Comment (stephlewyh) [inline src/seedu/addressbook/commands/AddCommand.java:70]: Remember to update with header comments as the overriden method has a different behavior from the default implementation.

File: src/seedu/addressbook/commands/Command.java [hunk 2]
```diff
@@ -70,4 +70,6 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating() { return false; }
```
Comment (stephlewyh) [inline src/seedu/addressbook/commands/Command.java:74]: This is a coding violation. Consider putting your `return` statement to the next line. :)

Comment (stephlewyh) [inline src/seedu/addressbook/commands/Command.java:74]: Having a default implementation of `isMutating()` in the parent class is good, as this reduces code. However, giving an arbitrary return value (E.g. true or false) reduces code understandability. Think about how refactoring to an `abstract` class with an  `abstract` method might work in this scenario.

File: src/seedu/addressbook/logic/Logic.java [hunk 3]
```diff
@@ -85,7 +87,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
```
Comment (stephlewyh) [inline src/seedu/addressbook/logic/Logic.java:87]: Remember to update with header comments, now that the behavior of `execute(Command command) `has changed.

**Review by stephlewyh — COMMENTED** [review]
Please note that going forward you need to update the tests as well when you make any behaviour changes.

File: src/seedu/addressbook/commands/AddCommand.java (hunk 1 above)
Comment (tristyxxnana) [reply src/seedu/addressbook/commands/AddCommand.java:70]: noted with thanks.

File: src/seedu/addressbook/commands/Command.java (hunk 2 above)
Comment (tristyxxnana) [reply src/seedu/addressbook/commands/Command.java:74]: Will take note on that. Thank you.

Comment (tristyxxnana) [reply src/seedu/addressbook/commands/Command.java:74]: I understand. Thank you.

File: src/seedu/addressbook/logic/Logic.java (hunk 3 above)
Comment (tristyxxnana) [reply src/seedu/addressbook/logic/Logic.java:87]: Alright I will do so in the upcoming tasks. Thank you.

**Pull request closed without merging**